pallets/merkle/src/lib.rs

```diff
@@ -571,6 +601,45 @@ impl<T: Config> Tree<T> for Pallet<T> {
 		Ok(tree_id)
 	}
 
+	fn initialize_tree(tree_id: T::TreeId, key_id: T::KeyId) -> Result<(), DispatchError> {
+		let mut tree = Trees::<T>::get(tree_id).ok_or(Error::<T>::TreeDoesntExist)?;
+		ensure!(!tree.initialized, Error::<T>::AlreadyInitialized);
+		let params = Self::get_verifying_key(key_id)?;
+		let (zero_tree, root_hash) = tree.setup.generate_zero_tree::<T>(tree.depth as usize, &params)?;
```

Comment:
We are saving the verifying keys and bp_gens in the same storage item. It would be easier to understand if they were separate.

Reply:
Since the bp_gens are basically the BP version of verifying keys, what are your thoughts on creating a `VerifyingKeyMetadata` such as
```
VerifyingKeyMetadata {
    data: Vec<u8>,
    type: VerifyingKeyType // (Bulletproof, Arkworks, etc)
}
```